drivers/spi/spi_ll_stm32.c

```diff
@@ -835,6 +837,17 @@ static int spi_stm32_init(const struct device *dev)
 		return -EIO;
 	}
 
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	if (spi == SPI1 || spi == SPI2 || spi == SPI3) {
+		/* SPI 1, 2, and 3 defaults clock source to PLL1
+		 * where the drivers expects the PCLK. Other sources
+		 * than the PCLK cannot be used as they are not defined
+		 * in `stm32_clock.h`, and thus not selectable in the DTS.
+		 */
+		LL_RCC_SetSPIClockSource(LL_RCC_SPI123_CLKSOURCE_CLKP);
```

Comment:
As shown in the picture you copied from Reference Manual, `LL_RCC_SPI123_CLKSOURCE_CLKP `doesn't correspond to PCLK , but it is the selection of `per_ck` .
 `per_ck`  is configurable, but none of the possibilities correspond to PCLK.
![image](https://user-images.githubusercontent.com/50730894/145549953-9a8bee4a-bf13-45d8-b77f-ca060ee8cbe4.png)

So PCLK is not a possible clock for SPI1/2/3. Thus it may be required to support other clock than PCLK in order to have SPI1/2/3 on H7